### ISSUE TITLE
1.0.2 using tar-ball does not build on RHEL 7.3

### ISSUE THREAD

**Issue by PaulKlumpp** [Dridi/libvmod-querystring#33]
Two points.

When using bootstrap:
* automake-version required is 1.15, but RHEL 7.3 does only have automake 1.13.x. Is automake 1.15 really needed?

When just doing configure, make:
* GCC outputs "-Wint-conversion" is an unknown option. So, same thing there. You may require a newer gcc that knows this option, but .. is the option really needed?

RHEL 7.3 is fully updated. Yea, RedHat is slow on advancing versions, but 1.0.1 builds on RHEL 7.3 just fine.

**Comment by Dridi** [comment]
Thank you for reporting, this is indeed unfortunate and the reasons why I missed that one are:

- I don't have (yet) the degree of portability I want in this project
- EPEL7 doesn't ship a recent-enough Varnish, so I couldn't *just* run a mock(1) build

To answer your questions:

> automake-version required is 1.15, but RHEL 7.3 does only have automake 1.13.x. Is automake 1.15 really needed?

I don't remember for sure, but since the release tarballs don't rely on automake, it should be possible to build a release (but not build from git by default).

> GCC outputs "-Wint-conversion" is an unknown option. So, same thing there. You may require a newer gcc that knows this option, but .. is the option really needed?

No, it's rather useful for development, especially after changes made in Varnish 5.1.1 that would trigger this warning. I can make it optional in a release dedicated to fixing this issue.

> RHEL 7.3 is fully updated. Yea, RedHat is slow on advancing versions, but 1.0.1 builds on RHEL 7.3 just fine.

Then you are good to go, 1.0.2 is basically the same as 1.0.1 but wouldn't build against the latest Varnish release (as described in the [release notes](https://github.com/Dridi/libvmod-querystring/releases/tag/v1.0.2)). Which reminds me that you didn't mention which version of Varnish you are building the VMOD against.

If you really want to build 1.0.2 this should work around GCC not recognizing the warning:

```bash
sed -i s/-Wint-conversion/-W/ configure
./configure
make
```

I haven't tested myself, but in any case you should be fine with 1.0.1 since the VMOD itself is the same.

**Comment by Dridi** [comment]
Can you please check that you can successfully build a dist archive of the master branch?

It's a 2-step process, first on a "modern" system:

```bash
./bootstrap
make dist
```

Then take the dist tarball on the CentOS host:

```bash
tar xf vmod-querystring-1.0.2.tar.gz
cd vmod-querystring/
./configure
make
```

It worked for me.

**Comment by PaulKlumpp** [comment]
I have been building against 4.1.6.
On RHEL 7.3:
```
yum info varnish-devel.x86_64
Loaded plugins: priorities
105 packages excluded due to repository priority protections
Available Packages
Name        : varnish-devel
Arch        : x86_64
Version     : 4.1.6
Release     : 1.el7
Size        : 76 k
Repo        : varnish-4.1-x86_64
Summary     : Development files for varnish
URL         : https://www.varnish-cache.org/
License     : BSD
Description : Development files for varnish-libs
            : Varnish Cache is a high-performance HTTP accelerator
```

**Comment by Dridi** [comment]
Hello @PaulKlumpp, since this issue was closed by a commit I forgot about it. Did you manage to build the VMOD for CentOS?

If not, I'm ready to release 1.0.3 but before that I'd like you to try first. You can grab the release archive from and let me know, in which case I'll go ahead and tag the release.

https://dridi.fedorapeople.org/vmod-querystring-1.0.3.tar.gz

**Comment by gauthier-delacroix** [comment]
@Dridi I've just tried your 1.0.3 tar and it compiles fine on CentOS7.

**Comment by Dridi** [comment]
@gauthier-delacroix thanks for confirming, I just released 1.0.3 that was ready to push a week ago. The release archive is available on github now.

**Comment by gauthier-delacroix** [comment]
@Dridi Thanks for the release (I get the `Say 33` joke ; I don't know what what does 'rounded R languages' doctors ask when looking in people throat :face_with_thermometer:)
